# Amsterda urban studies

In [2]:
import pandas as pd
import requests

## APIs

### Energy label

#### Endpoint

#### Response

In [1]:
dict = {
    "_embedded": {
        "energielabel": [
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1011PG~171~~~~0363010001011891~",
                        "title": "20160601~1011PG~171~~~~0363010001011891~",
                        "id": "20160601~1011PG~171~~~~0363010001011891~"
                    }
                },
                "id": "20160601~1011PG~171~~~~0363010001011891~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "1Rgielabel.nl",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Maisonnette",
                "gebouwsubtype": "Tussenmidden",
                "sbicode": "",
                "postcode": "1011PG",
                "huisnummer": 171,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010001011891",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 2002,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "B",
                "energieindex": "1,21",
                "energieindexMetEmgForfaitair": "1,21",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "2796,08",
                "berekendeEnergieverbruik": "52754,00"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1015JJ~7~2~~~0363010000647799~",
                        "title": "20160601~1015JJ~7~2~~~0363010000647799~",
                        "id": "20160601~1015JJ~7~2~~~0363010000647799~"
                    }
                },
                "id": "20160601~1015JJ~7~2~~~0363010000647799~",
                "registratiedatum": "2016-06-02",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Innax Bouwkundig adviseurs BV",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Tussenmidden",
                "sbicode": "",
                "postcode": "1015JJ",
                "huisnummer": 7,
                "huisletter": "",
                "huisnummertoevoeging": "2",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000647799",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1905,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "B",
                "energieindex": "1,39",
                "energieindexMetEmgForfaitair": "1,39",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "1883,73",
                "berekendeEnergieverbruik": "36381,54"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1015NG~177~1~~~0363010000954233~",
                        "title": "20160601~1015NG~177~1~~~0363010000954233~",
                        "id": "20160601~1015NG~177~1~~~0363010000954233~"
                    }
                },
                "id": "20160601~1015NG~177~1~~~0363010000954233~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Eneco installatiebedrijven",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Rekenmethodiek Definitief Energielabel, versie 1.2, 16 september 2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Maisonnette",
                "gebouwsubtype": "Tussendak",
                "sbicode": "",
                "postcode": "1015NG",
                "huisnummer": 177,
                "huisletter": "",
                "huisnummertoevoeging": "1",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000954233",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1930,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "F",
                "energieindex": "",
                "energieindexMetEmgForfaitair": "",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "0,00",
                "berekendeEnergieverbruik": "116814,21"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1015XT~48~H~~~0363010000736175~",
                        "title": "20160601~1015XT~48~H~~~0363010000736175~",
                        "id": "20160601~1015XT~48~H~~~0363010000736175~"
                    }
                },
                "id": "20160601~1015XT~48~H~~~0363010000736175~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Mijnbouwhulp.nl",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Rekenmethodiek Definitief Energielabel, versie 1.2, 16 september 2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Tussenvloer",
                "sbicode": "",
                "postcode": "1015XT",
                "huisnummer": 48,
                "huisletter": "",
                "huisnummertoevoeging": "H",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000736175",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1930,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "D",
                "energieindex": "",
                "energieindexMetEmgForfaitair": "",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "0,00",
                "berekendeEnergieverbruik": "67499,07"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1018NM~144~~A~~0363010012094717~",
                        "title": "20160601~1018NM~144~~A~~0363010012094717~",
                        "id": "20160601~1018NM~144~~A~~0363010012094717~"
                    }
                },
                "id": "20160601~1018NM~144~~A~~0363010012094717~",
                "registratiedatum": "2016-06-14",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woningstichting Eigen Haard",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Maisonnette",
                "gebouwsubtype": "Tussenvloer",
                "sbicode": "",
                "postcode": "1018NM",
                "huisnummer": 144,
                "huisletter": "A",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010012094717",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1880,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "1,13",
                "energieindexMetEmgForfaitair": "1,13",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "1801,83",
                "berekendeEnergieverbruik": "33671,96"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1018NM~144~~C~~0363010012094719~",
                        "title": "20160601~1018NM~144~~C~~0363010012094719~",
                        "id": "20160601~1018NM~144~~C~~0363010012094719~"
                    }
                },
                "id": "20160601~1018NM~144~~C~~0363010012094719~",
                "registratiedatum": "2016-06-14",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woningstichting Eigen Haard",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Tussenmidden",
                "sbicode": "",
                "postcode": "1018NM",
                "huisnummer": 144,
                "huisletter": "C",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010012094719",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1880,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "1,13",
                "energieindexMetEmgForfaitair": "1,13",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "1289,54",
                "berekendeEnergieverbruik": "23723,78"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1018NM~144~~D~~0363010012094720~",
                        "title": "20160601~1018NM~144~~D~~0363010012094720~",
                        "id": "20160601~1018NM~144~~D~~0363010012094720~"
                    }
                },
                "id": "20160601~1018NM~144~~D~~0363010012094720~",
                "registratiedatum": "2016-06-14",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woningstichting Eigen Haard",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Maisonnette",
                "gebouwsubtype": "Tussendak",
                "sbicode": "",
                "postcode": "1018NM",
                "huisnummer": 144,
                "huisletter": "D",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010012094720",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1880,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "1,01",
                "energieindexMetEmgForfaitair": "1,01",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "1725,53",
                "berekendeEnergieverbruik": "32151,12"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1018NN~148~~B~~0363010012094723~",
                        "title": "20160601~1018NN~148~~B~~0363010012094723~",
                        "id": "20160601~1018NN~148~~B~~0363010012094723~"
                    }
                },
                "id": "20160601~1018NN~148~~B~~0363010012094723~",
                "registratiedatum": "2016-06-14",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woningstichting Eigen Haard",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Tussenmidden",
                "sbicode": "",
                "postcode": "1018NN",
                "huisnummer": 148,
                "huisletter": "B",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010012094723",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1880,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "1,10",
                "energieindexMetEmgForfaitair": "1,10",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "1801,46",
                "berekendeEnergieverbruik": "33477,03"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1018NN~148~~D~~0363010012094725~",
                        "title": "20160601~1018NN~148~~D~~0363010012094725~",
                        "id": "20160601~1018NN~148~~D~~0363010012094725~"
                    }
                },
                "id": "20160601~1018NN~148~~D~~0363010012094725~",
                "registratiedatum": "2016-06-14",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woningstichting Eigen Haard",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Maisonnette",
                "gebouwsubtype": "Tussendak",
                "sbicode": "",
                "postcode": "1018NN",
                "huisnummer": 148,
                "huisletter": "D",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010012094725",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1880,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "1,07",
                "energieindexMetEmgForfaitair": "1,07",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "2942,61",
                "berekendeEnergieverbruik": "55266,27"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1018TZ~44~~~~0363010000673388~",
                        "title": "20160601~1018TZ~44~~~~0363010000673388~",
                        "id": "20160601~1018TZ~44~~~~0363010000673388~"
                    }
                },
                "id": "20160601~1018TZ~44~~~~0363010000673388~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Energielabel Hofstad",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Rekenmethodiek Definitief Energielabel, versie 1.2, 16 september 2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Hoekdak",
                "sbicode": "",
                "postcode": "1018TZ",
                "huisnummer": 44,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000673388",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1985,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "C",
                "energieindex": "",
                "energieindexMetEmgForfaitair": "",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "0,00",
                "berekendeEnergieverbruik": "74612,91"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1019GW~41~~~~0363010001162986~",
                        "title": "20160601~1019GW~41~~~~0363010001162986~",
                        "id": "20160601~1019GW~41~~~~0363010001162986~"
                    }
                },
                "id": "20160601~1019GW~41~~~~0363010001162986~",
                "registratiedatum": "2016-06-03",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Innax Bouwkundig adviseurs BV",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Maisonnette",
                "gebouwsubtype": "Tussendak",
                "sbicode": "",
                "postcode": "1019GW",
                "huisnummer": 41,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010001162986",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 2004,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "0,93",
                "energieindexMetEmgForfaitair": "0,93",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "3522,95",
                "berekendeEnergieverbruik": "57470,56"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1019HD~105~~~~0363010001038167~",
                        "title": "20160601~1019HD~105~~~~0363010001038167~",
                        "id": "20160601~1019HD~105~~~~0363010001038167~"
                    }
                },
                "id": "20160601~1019HD~105~~~~0363010001038167~",
                "registratiedatum": "2016-06-09",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Duresta Real Estate Solutions BV",
                "opnametype": "",
                "status": "",
                "berekeningstype": "ISSO75.3, versie 3.0, oktober 2011",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "U",
                "gebouwtype": "",
                "gebouwsubtype": "",
                "sbicode": "81",
                "postcode": "1019HD",
                "huisnummer": 105,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010001038167",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 2006,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "0,92",
                "energieindexMetEmgForfaitair": "0,92",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "45,85",
                "berekendeEnergieverbruik": "610,20"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1019LG~465~~~~0363010000953261~",
                        "title": "20160601~1019LG~465~~~~0363010000953261~",
                        "id": "20160601~1019LG~465~~~~0363010000953261~"
                    }
                },
                "id": "20160601~1019LG~465~~~~0363010000953261~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Energetisch- en Bouwkundig Adviesbureau Peters",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Rekenmethodiek Definitief Energielabel, versie 1.2, 16 september 2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Hoekmidden",
                "sbicode": "",
                "postcode": "1019LG",
                "huisnummer": 465,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000953261",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1995,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "B",
                "energieindex": "",
                "energieindexMetEmgForfaitair": "",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "0,00",
                "berekendeEnergieverbruik": "52349,89"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1025HT~531~~~~0363010000666241~",
                        "title": "20160601~1025HT~531~~~~0363010000666241~",
                        "id": "20160601~1025HT~531~~~~0363010000666241~"
                    }
                },
                "id": "20160601~1025HT~531~~~~0363010000666241~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "De Label Expert",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Rekenmethodiek Definitief Energielabel, versie 1.2, 16 september 2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Hoekmidden",
                "sbicode": "",
                "postcode": "1025HT",
                "huisnummer": 531,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000666241",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1970,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "F",
                "energieindex": "",
                "energieindexMetEmgForfaitair": "",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "0,00",
                "berekendeEnergieverbruik": "94056,68"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1025MS~54~~~~0363010000981826~",
                        "title": "20160601~1025MS~54~~~~0363010000981826~",
                        "id": "20160601~1025MS~54~~~~0363010000981826~"
                    }
                },
                "id": "20160601~1025MS~54~~~~0363010000981826~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Labelnu",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Rekenmethodiek Definitief Energielabel, versie 1.2, 16 september 2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Rijwoning tussen",
                "gebouwsubtype": "",
                "sbicode": "",
                "postcode": "1025MS",
                "huisnummer": 54,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000981826",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1995,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "B",
                "energieindex": "",
                "energieindexMetEmgForfaitair": "",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "0,00",
                "berekendeEnergieverbruik": "73880,56"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1033EK~10~~~~0363010000880303~",
                        "title": "20160601~1033EK~10~~~~0363010000880303~",
                        "id": "20160601~1033EK~10~~~~0363010000880303~"
                    }
                },
                "id": "20160601~1033EK~10~~~~0363010000880303~",
                "registratiedatum": "2016-06-06",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woning Advies Nederland B.V.",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Maisonnette",
                "gebouwsubtype": "Tussenmidden",
                "sbicode": "",
                "postcode": "1033EK",
                "huisnummer": 10,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000880303",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1922,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "D",
                "energieindex": "1,87",
                "energieindexMetEmgForfaitair": "1,87",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "4181,87",
                "berekendeEnergieverbruik": "80916,65"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1034NM~42~~~~0363010000893669~",
                        "title": "20160601~1034NM~42~~~~0363010000893669~",
                        "id": "20160601~1034NM~42~~~~0363010000893669~"
                    }
                },
                "id": "20160601~1034NM~42~~~~0363010000893669~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woningstichting Rochdale",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Hoekvloer",
                "sbicode": "",
                "postcode": "1034NM",
                "huisnummer": 42,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000893669",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1977,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "C",
                "energieindex": "1,50",
                "energieindexMetEmgForfaitair": "1,50",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "2673,24",
                "berekendeEnergieverbruik": "50582,99"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1036KL~26~~~~0363010011211107~",
                        "title": "20160601~1036KL~26~~~~0363010011211107~",
                        "id": "20160601~1036KL~26~~~~0363010011211107~"
                    }
                },
                "id": "20160601~1036KL~26~~~~0363010011211107~",
                "registratiedatum": "2016-06-01",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Het Labelhuis",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Rekenmethodiek Definitief Energielabel, versie 1.2, 16 september 2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Rijwoning tussen",
                "gebouwsubtype": "",
                "sbicode": "",
                "postcode": "1036KL",
                "huisnummer": 26,
                "huisletter": "",
                "huisnummertoevoeging": "",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010011211107",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 2010,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "A",
                "energieindex": "",
                "energieindexMetEmgForfaitair": "",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "0,00",
                "berekendeEnergieverbruik": "61687,38"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1051HZ~10~1~~~0363010000845715~",
                        "title": "20160601~1051HZ~10~1~~~0363010000845715~",
                        "id": "20160601~1051HZ~10~1~~~0363010000845715~"
                    }
                },
                "id": "20160601~1051HZ~10~1~~~0363010000845715~",
                "registratiedatum": "2016-09-16",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woning Advies Nederland B.V.",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Tussenmidden",
                "sbicode": "",
                "postcode": "1051HZ",
                "huisnummer": 10,
                "huisletter": "",
                "huisnummertoevoeging": "1",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000845715",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1920,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "C",
                "energieindex": "1,52",
                "energieindexMetEmgForfaitair": "1,52",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "2312,57",
                "berekendeEnergieverbruik": "44367,32"
            },
            {
                "_links": {
                    "schema": "https://schemas.data.amsterdam.nl/datasets/duurzaamheid/energielabel/v1",
                    "self": {
                        "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel/20160601~1051HZ~10~3~~~0363010000845717~",
                        "title": "20160601~1051HZ~10~3~~~0363010000845717~",
                        "id": "20160601~1051HZ~10~3~~~0363010000845717~"
                    }
                },
                "id": "20160601~1051HZ~10~3~~~0363010000845717~",
                "registratiedatum": "2016-09-16",
                "opnamedatum": "2016-06-01",
                "metingGeldigTot": "2026-06-01",
                "certificaathouder": "Woning Advies Nederland B.V.",
                "opnametype": "",
                "status": "",
                "berekeningstype": "Nader Voorschrift, versie 1.0, 1 februari 2014 met erratalijst dd 03-11-2014",
                "isOpBasisVanReferentieGebouw": "",
                "gebouwklasse": "W",
                "gebouwtype": "Flatwoning (overig)",
                "gebouwsubtype": "Tussendak",
                "sbicode": "",
                "postcode": "1051HZ",
                "huisnummer": 10,
                "huisletter": "",
                "huisnummertoevoeging": "3",
                "detailaanduiding": "",
                "bagVerblijfsobjectId": "0363010000845717",
                "bagLigplaatsId": "",
                "bagStandplaatsId": "",
                "bagPandIds": "",
                "projectnaam": "",
                "projectobject": "",
                "bouwjaar": 1920,
                "gebruiksoppervlakteThermischeZone": "",
                "compactheid": "",
                "energieklasse": "C",
                "energieindex": "1,73",
                "energieindexMetEmgForfaitair": "1,73",
                "eisEnergiebehoefte": "",
                "primaireFossieleEnergie": "",
                "primaireFossieleEnergieEmgForfaitair": "",
                "aandeelHernieuwbareEnergie": "",
                "aandeelHernieuwbareEnergieEmgForfaitair": "",
                "temperatuuroverschrijding": "",
                "warmtebehoefte": "",
                "energiebehoefte": "",
                "eisPrimaireFossieleEnergie": "",
                "eisAandeelHernieuwbareEnergie": "",
                "eisTemperatuuroverschrijding": "",
                "berekendeCo2Emissie": "3117,34",
                "berekendeEnergieverbruik": "60233,39"
            }
        ]
    },
    "_links": {
        "self": {
            "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel"
        },
        "next": {
            "href": "https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel?page=2"
        }
    },
    "page": {
        "number": 1,
        "size": 20
    }
}

In [2]:
dict.keys()

dict_keys(['_embedded', '_links', 'page'])

In [6]:
dict['_embedded'].keys()

dict_keys(['energielabel'])

In [10]:
dict['_embedded']['energielabel'][0].keys()

dict_keys(['_links', 'id', 'registratiedatum', 'opnamedatum', 'metingGeldigTot', 'certificaathouder', 'opnametype', 'status', 'berekeningstype', 'isOpBasisVanReferentieGebouw', 'gebouwklasse', 'gebouwtype', 'gebouwsubtype', 'sbicode', 'postcode', 'huisnummer', 'huisletter', 'huisnummertoevoeging', 'detailaanduiding', 'bagVerblijfsobjectId', 'bagLigplaatsId', 'bagStandplaatsId', 'bagPandIds', 'projectnaam', 'projectobject', 'bouwjaar', 'gebruiksoppervlakteThermischeZone', 'compactheid', 'energieklasse', 'energieindex', 'energieindexMetEmgForfaitair', 'eisEnergiebehoefte', 'primaireFossieleEnergie', 'primaireFossieleEnergieEmgForfaitair', 'aandeelHernieuwbareEnergie', 'aandeelHernieuwbareEnergieEmgForfaitair', 'temperatuuroverschrijding', 'warmtebehoefte', 'energiebehoefte', 'eisPrimaireFossieleEnergie', 'eisAandeelHernieuwbareEnergie', 'eisTemperatuuroverschrijding', 'berekendeCo2Emissie', 'berekendeEnergieverbruik'])

#### Potential hypothesis

In [11]:
dict['_links'].keys()

dict_keys(['self', 'next'])

In [12]:
dict['_links']['self']

{'href': 'https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel'}

In [13]:
dict['_links']['next']

{'href': 'https://api.data.amsterdam.nl/v1/duurzaamheid/energielabel?page=2'}

### Municipal real estate - Gemeentelijk vastgoed

#### Endpoint

https://data.amsterdam.nl/catalogus/65

In [45]:
url="https://api.data.amsterdam.nl/v1/gemeentelijk_vastgoed/v2/gebouwobjecten"
response_municipal = requests.get(url)
print(response_municipal.status_code)

200


#### Response

In [46]:
response_municipal.content

b'{\n  "_embedded":{\n  "gebouwobjecten":[\n  {"_links":{"schema":"https://schemas.data.amsterdam.nl/datasets/gemeentelijk_vastgoed/gebouwobjecten/v1","self":{"href":"https://api.data.amsterdam.nl/v1/gemeentelijk_vastgoed/v2/gebouwobjecten/339","title":"339","objectId":339}},"objectId":339,"straatnaam":"Aldebaranplein","huisnummer":null,"huisletter":null,"huisnummertoevoeging":"2-4","huisnummeraanduiding":null,"eigendom":"Volle eigendom","bagPandIds":["0363100012062051"],"geometrie":{"type":"MultiPolygon","coordinates":[[[[121372.063,491634.878],[121376.943,491631.063],[121391.455,491649.607],[121387.537,491652.685],[121388.672,491654.135],[121384.725,491657.232],[121384.585,491657.342],[121372.794,491642.255],[121370.318,491644.189],[121369.794,491643.519],[121367.381,491645.403],[121367.311,491645.314],[121356.804,491631.911],[121354.963,491633.353],[121353.057,491630.911],[121345.047,491637.161],[121344.739,491636.744],[121344.487,491636.933],[121344.22,491637.099],[121343.94,491637

In [47]:
info_municipal = response.json()
info_municipal

{'_embedded': {'gebouwobjecten': [{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/gemeentelijk_vastgoed/gebouwobjecten/v1',
     'self': {'href': 'https://api.data.amsterdam.nl/v1/gemeentelijk_vastgoed/v2/gebouwobjecten/339',
      'title': '339',
      'objectId': 339}},
    'objectId': 339,
    'straatnaam': 'Aldebaranplein',
    'huisnummer': None,
    'huisletter': None,
    'huisnummertoevoeging': '2-4',
    'huisnummeraanduiding': None,
    'eigendom': 'Volle eigendom',
    'bagPandIds': ['0363100012062051'],
    'geometrie': {'type': 'MultiPolygon',
     'coordinates': [[[[121372.063, 491634.878],
        [121376.943, 491631.063],
        [121391.455, 491649.607],
        [121387.537, 491652.685],
        [121388.672, 491654.135],
        [121384.725, 491657.232],
        [121384.585, 491657.342],
        [121372.794, 491642.255],
        [121370.318, 491644.189],
        [121369.794, 491643.519],
        [121367.381, 491645.403],
        [121367.311, 491645.31

In [48]:
type(info_municipal)

dict

In [49]:
info_municipal.keys()

dict_keys(['_embedded', '_links', 'page'])

In [50]:
info_municipal['_embedded'].keys()

dict_keys(['gebouwobjecten'])

In [51]:
info_municipal['_embedded']['gebouwobjecten'][0].keys()

dict_keys(['_links', 'objectId', 'straatnaam', 'huisnummer', 'huisletter', 'huisnummertoevoeging', 'huisnummeraanduiding', 'eigendom', 'bagPandIds', 'geometrie'])

#### Finding

**No postcode**

### Building -gebouwobjecten

#### Endpoint

In [58]:
url_buildingobject="https://api.data.amsterdam.nl/v1/gemeentelijk_vastgoed/v2/gebouwobjecten"
response_buildingobject = requests.get(url_buildingobject)
print(response_buildingobject.status_code)

200


#### Response

In [59]:
info_buildingobject = response_buildingobject.json()
info_buildingobject

{'_embedded': {'gebouwobjecten': [{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/gemeentelijk_vastgoed/gebouwobjecten/v1',
     'self': {'href': 'https://api.data.amsterdam.nl/v1/gemeentelijk_vastgoed/v2/gebouwobjecten/339',
      'title': '339',
      'objectId': 339}},
    'objectId': 339,
    'straatnaam': 'Aldebaranplein',
    'huisnummer': None,
    'huisletter': None,
    'huisnummertoevoeging': '2-4',
    'huisnummeraanduiding': None,
    'eigendom': 'Volle eigendom',
    'bagPandIds': ['0363100012062051'],
    'geometrie': {'type': 'MultiPolygon',
     'coordinates': [[[[121372.063, 491634.878],
        [121376.943, 491631.063],
        [121391.455, 491649.607],
        [121387.537, 491652.685],
        [121388.672, 491654.135],
        [121384.725, 491657.232],
        [121384.585, 491657.342],
        [121372.794, 491642.255],
        [121370.318, 491644.189],
        [121369.794, 491643.519],
        [121367.381, 491645.403],
        [121367.311, 491645.31

In [60]:
info_buildingobject.keys()

dict_keys(['_embedded', '_links', 'page'])

In [65]:
info_buildingobject['_embedded']['gebouwobjecten'][0]

{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/gemeentelijk_vastgoed/gebouwobjecten/v1',
  'self': {'href': 'https://api.data.amsterdam.nl/v1/gemeentelijk_vastgoed/v2/gebouwobjecten/339',
   'title': '339',
   'objectId': 339}},
 'objectId': 339,
 'straatnaam': 'Aldebaranplein',
 'huisnummer': None,
 'huisletter': None,
 'huisnummertoevoeging': '2-4',
 'huisnummeraanduiding': None,
 'eigendom': 'Volle eigendom',
 'bagPandIds': ['0363100012062051'],
 'geometrie': {'type': 'MultiPolygon',
  'coordinates': [[[[121372.063, 491634.878],
     [121376.943, 491631.063],
     [121391.455, 491649.607],
     [121387.537, 491652.685],
     [121388.672, 491654.135],
     [121384.725, 491657.232],
     [121384.585, 491657.342],
     [121372.794, 491642.255],
     [121370.318, 491644.189],
     [121369.794, 491643.519],
     [121367.381, 491645.403],
     [121367.311, 491645.314],
     [121356.804, 491631.911],
     [121354.963, 491633.353],
     [121353.057, 491630.911],
     [121

### Markets

#### Endpoint

In [ ]:
404

### Basic Register of Addresses and Buildings - (BAG) Basisregistratie adressen en gebouwen (BAG)

#### Endpoint

In [68]:
url_bag="https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres"
response_bag = requests.get(url_bag)
print(response_bag.status_code)

200


#### Response

In [69]:
info_bag = response_bag.json()
info_bag

{'_embedded': {'verblijfsobjecten': [{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/bag/verblijfsobjecten/v1',
     'self': {'href': 'https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten/0363010000543292?volgnummer=1',
      'title': '0363010000543292',
      'identificatie': '0363010000543292',
      'volgnummer': 1},
     'heeftHoofdadres': {'href': 'https://api.data.amsterdam.nl/v1/bag/nummeraanduidingen/0363200000006110?volgnummer=1',
      'title': '0363200000006110',
      'identificatie': '0363200000006110',
      'volgnummer': 1},
     'ligtInBuurt': {'href': 'https://api.data.amsterdam.nl/v1/gebieden/buurten/03630980000029?volgnummer=1',
      'title': 'Anjeliersbuurt-Zuid',
      'identificatie': '03630980000029',
      'volgnummer': 1},
     'heeftDossier': None,
     'heeftNevenadres': [],
     'ligtInPanden': [{'href': 'https://api.data.amsterdam.nl/v1/bag/panden/0363100012174784?volgnummer=4',
       'title': '0363100012174784',
       'identificatie'

In [70]:
info_bag['_embedded'].keys()

dict_keys(['verblijfsobjecten', 'heeftHoofdadres'])

In [71]:
info_bag['_embedded']['verblijfsobjecten'][0].keys()

dict_keys(['_links', 'identificatie', 'volgnummer', 'plusvolgnummer', 'registratiedatum', 'cbsNummer', 'indicatieWoningvoorraad', 'financieringscodeCode', 'financieringscodeOmschrijving', 'geconstateerd', 'heeftHoofdadresId', 'geometrie', 'oppervlakte', 'statusCode', 'statusOmschrijving', 'beginGeldigheid', 'eindGeldigheid', 'documentdatum', 'documentnummer', 'gebruiksdoelWoonfunctieCode', 'gebruiksdoelWoonfunctieOmschrijving', 'gebruiksdoelGezondheidszorgfunctieCode', 'gebruiksdoelGezondheidszorgfunctieOmschrijving', 'aantalEenhedenComplex', 'verdiepingToegang', 'aantalBouwlagen', 'hoogsteBouwlaag', 'laagsteBouwlaag', 'aantalKamers', 'eigendomsverhoudingCode', 'eigendomsverhoudingOmschrijving', 'feitelijkGebruikCode', 'feitelijkGebruikOmschrijving', 'redenopvoerCode', 'redenopvoerOmschrijving', 'redenafvoerCode', 'redenafvoerOmschrijving', 'ligtInBuurtId', 'heeftDossierId', 'bagprocesCode', 'bagprocesOmschrijving', 'datumActueelTot', 'gebruiksdoel', 'toegang'])

In [72]:
info_bag['_embedded']['heeftHoofdadres'][0].keys()

dict_keys(['_links', 'identificatie', 'volgnummer', 'plusvolgnummer', 'registratiedatum', 'huisnummer', 'geconstateerd', 'huisletter', 'huisnummertoevoeging', 'postcode', 'ligtInWoonplaatsId', 'beginGeldigheid', 'eindGeldigheid', 'ligtAanOpenbareruimteId', 'typeAdresseerbaarObjectCode', 'typeAdresseerbaarObjectOmschrijving', 'documentdatum', 'documentnummer', 'statusCode', 'statusOmschrijving', 'typeAdres', 'adresseertVerblijfsobjectId', 'adresseertLigplaatsId', 'adresseertStandplaatsId', 'heeftDossierId', 'bagprocesCode', 'bagprocesOmschrijving', 'datumActueelTot'])

#### Iteration through pages

In [82]:

# This variable starts with the same value as we had for url_bag,
# but will change as the code goes throuh the pages:
headers = {
    "X-Api-Key": "eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI0OTk2Mzk2MTk0MTcyNjEwNjM1IiwiZXhwIjoxODE0MzU1ODQ1fQ.LfXJNucSt2nDBrgA60ElAZdpz89--auFrCQbU3Re17e-MYHeZhBPc34x0nVxc-Ri4xCB0LuiST2UDmZnaOnnAg"
}
url_pagination="https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres"

all_records_heeftHoofdadres = []
page = 1

while url_pagination:

    print(f"Downloading page {page}")

    response_page = requests.get(url_pagination)
    response_page.raise_for_status()

    info_page = response_page.json()

    # Add the records from this page
    all_records_heeftHoofdadres.extend(info_page["_embedded"]["heeftHoofdadres"])

    # Get the URL for the next page (if any)
    url_pagination = info_page["_links"].get("next", {}).get("href")

    page += 1
    

HTTPError: 403 Client Error: Forbidden for url: https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres&page=101

In [102]:
info_page["_links"]

{'self': {'href': 'https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres&page=100'},
 'next': {'href': 'https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres&page=101'},
 'previous': {'href': 'https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres&page=99'}}

In [ ]:

url_pagination

In [116]:
headers = {
    "X-Api-Key": "eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI0OTk2Mzk2MTk0MTcyNjEwNjM1IiwiZXhwIjoxODE0MzU1ODQ1fQ.LfXJNucSt2nDBrgA60ElAZdpz89--auFrCQbU3Re17e-MYHeZhBPc34x0nVxc-Ri4xCB0LuiST2UDmZnaOnnAg"
}

url_test_pagination="https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres&page=101"
url_test2_pagination="https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?_expandScope=heeftHoofdadres&_pageSize=100&_count=true"
url_test3_pagination="https://api.data.amsterdam.nl/v1/bag/verblijfsobjecten?page=101"


response_test_page = requests.get(url_test3_pagination, headers=headers)
print(response_test_page.status_code)

403


In [110]:
response_test_page_object= response_test_page.json()

In [111]:
response_test_page_object['page']['totalPages']

5820

In [112]:
response_test_page_object["page"]["totalElements"]

581965

#### Transform to Data Frame

In [83]:
len(all_records_heeftHoofdadres)

2000

In [84]:
municipal_bag_data = pd.DataFrame(all_records_heeftHoofdadres)

In [85]:
municipal_bag_data

,_links,identificatie,volgnummer,plusvolgnummer,registratiedatum,huisnummer,geconstateerd,huisletter,huisnummertoevoeging,postcode,...,statusCode,statusOmschrijving,typeAdres,adresseertVerblijfsobjectId,adresseertLigplaatsId,adresseertStandplaatsId,heeftDossierId,bagprocesCode,bagprocesOmschrijving,datumActueelTot
0,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000006110,1,3,2010-11-04T21:57:01,1,False,NaN,H,1015NR,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000543292,None,None,None,None,None,None
1,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000006111,1,3,2010-11-04T21:57:01,1,False,NaN,1,1015NR,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000543293,None,None,None,None,None,None
2,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000006112,1,3,2010-11-04T21:57:01,1,False,NaN,2,1015NR,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000543294,None,None,None,None,None,None
3,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000006114,1,3,2010-11-04T21:57:02,3,False,NaN,1,1015NR,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000543296,None,None,None,None,None,None
4,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000006115,1,3,2010-11-04T21:57:02,3,False,NaN,2,1015NR,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000543297,None,None,None,None,None,None
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1995,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000008848,1,3,2010-11-04T22:03:08,46,False,A,NaN,1072NM,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000546003,None,None,None,None,None,None
1996,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000008849,1,3,2010-11-04T22:03:08,46,False,B,NaN,1072NM,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000546004,None,None,None,None,None,None
1997,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000008850,1,3,2010-11-04T22:03:08,46,False,C,NaN,1072NM,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000546005,None,None,None,None,None,None
1998,{'schema': 'https://schemas.data.amsterdam.nl/...,0363200000008851,1,3,2010-11-04T22:03:08,46,False,D,NaN,1072NM,...,None,Naamgeving uitgegeven,Hoofdadres,0363010000546006,None,None,None,None,None,None


In [87]:
municipal_bag_data['postcode'].unique()

<StringArray>
['1015NR', '1094JX', '1094KL', '1069LK', '1094JZ', '1022XE', '1094KA',
 '1094KB', '1094KC', '1094KD',
 ...
 '1072BD', '1072BH', '1016VR', '1072NK', '1072NA', '1091ED', '1072NL',
 '1072NB', '1053EK', '1072NM']
Length: 126, dtype: str

In [89]:
found_postCode = municipal_bag_data['postcode'].unique()

In [92]:
type(found_postCode[1])

str

In [95]:
found_postCode = [
    postcode[:4] if isinstance(postcode, str) else postcode
    for postcode in found_postCode
]
found_postCode

['1015',
 '1094',
 '1094',
 '1069',
 '1094',
 '1022',
 '1094',
 '1094',
 '1094',
 '1094',
 '1094',
 '1052',
 '1094',
 '1094',
 '1094',
 '1094',
 '1094',
 '1094',
 '1094',
 '1094',
 '1016',
 '1016',
 '1011',
 '1013',
 '1073',
 '1091',
 '1091',
 '1091',
 '1073',
 '1052',
 '1015',
 '1052',
 '1013',
 '1013',
 '1092',
 '1018',
 '1095',
 '1095',
 '1018',
 '1018',
 '1054',
 '1054',
 '1017',
 '1011',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1031',
 '1015',
 '1073',
 '1015',
 '1013',
 '1066',
 '1015',
 '1054',
 '1054',
 '1054',
 '1015',
 '1052',
 '1074',
 '1019',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1012',
 '1054',
 '1012',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1054',
 '1102',
 '1051',
 '1054',
 '1054',
 '1054',
 '1054',
 '1052',
 '1052',
 '1052',
 '1052',
 '1052',
 '1052',
 '1016',
 '1052',
 '1052',
 

In [100]:
unique_postcodes = set(found_postCode)
len(unique_postcodes)

25

In [101]:
unique_postcodes 

{'1011',
 '1012',
 '1013',
 '1015',
 '1016',
 '1017',
 '1018',
 '1019',
 '1022',
 '1031',
 '1051',
 '1052',
 '1053',
 '1054',
 '1066',
 '1069',
 '1072',
 '1073',
 '1074',
 '1091',
 '1092',
 '1094',
 '1095',
 '1102',
 nan}

### Trash

Distance from homes to trash deposits - Loopafstanden

In [ ]:
Service areas - trash

In [ ]:
https://data.amsterdam.nl/catalogus/55

#### Tests

In [122]:
url_test1_trash="https://api.data.amsterdam.nl/v1/huishoudelijkafval/container"
url_test2_trash="https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?&_pageSize=100&_count=true"
url_test3_trash="https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?fractieCode=2&_pageSize=100&_count=true"


response_test3_trash = requests.get(url_test3_trash)
print(response_test3_trash.status_code)
response_test3_trash_object= response_test3_trash.json()
response_test3_trash_object

200


{'_embedded': {'container': [{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/huishoudelijkafval/container/v2',
     'self': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/container/10004',
      'title': '10004',
      'id': '10004'},
     'cluster': None,
     'locatie': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containerlocatie/10003',
      'title': '10003',
      'id': '10003'},
     'type': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containertype/348',
      'title': '348',
      'id': '348'},
     'bagHoofdadresVerblijfsobject': None,
     'gbdBuurt': {'href': 'https://api.data.amsterdam.nl/v1/gebieden/buurten/03630980000386',
      'title': '03630980000386',
      'identificatie': '03630980000386'},
     'bagOpenbareruimte': None,
     'bagNummeraanduiding': None},
    'id': '10004',
    'idNummer': 'GL 7226',
    'serienummer': 'BAO01404',
    'clusterId': None,
    'eigenaarId': '184',
    'eigenaarNaam': 

In [121]:
response_test_trash_object["page"]['totalPages']

304

In [120]:
response_test_trash_object["page"]["totalElements"]

30365

In [123]:
response_test3_trash_object["page"]['totalPages']

31

In [124]:
response_test3_trash_object["page"]["totalElements"]

3047

#### Response

In [3]:
url_trash="https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?fractieCode=2"

response_trash = requests.get(url_trash)
print(response_trash.status_code)

info_trash = response_trash.json()
info_trash

200


{'_embedded': {'container': [{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/huishoudelijkafval/container/v2',
     'self': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/container/10004',
      'title': '10004',
      'id': '10004'},
     'cluster': None,
     'locatie': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containerlocatie/10003',
      'title': '10003',
      'id': '10003'},
     'type': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containertype/348',
      'title': '348',
      'id': '348'},
     'bagHoofdadresVerblijfsobject': None,
     'gbdBuurt': {'href': 'https://api.data.amsterdam.nl/v1/gebieden/buurten/03630980000386',
      'title': '03630980000386',
      'identificatie': '03630980000386'},
     'bagOpenbareruimte': None,
     'bagNummeraanduiding': None},
    'id': '10004',
    'idNummer': 'GL 7226',
    'serienummer': 'BAO01404',
    'clusterId': None,
    'eigenaarId': '184',
    'eigenaarNaam': 

In [4]:
info_trash['_embedded'].keys()

dict_keys(['container'])

In [8]:
len(info_trash['_embedded']['container'])

20

In [6]:
info_trash['_embedded']['container'][0]

{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/huishoudelijkafval/container/v2',
  'self': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/container/10004',
   'title': '10004',
   'id': '10004'},
  'cluster': None,
  'locatie': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containerlocatie/10003',
   'title': '10003',
   'id': '10003'},
  'type': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containertype/348',
   'title': '348',
   'id': '348'},
  'bagHoofdadresVerblijfsobject': None,
  'gbdBuurt': {'href': 'https://api.data.amsterdam.nl/v1/gebieden/buurten/03630980000386',
   'title': '03630980000386',
   'identificatie': '03630980000386'},
  'bagOpenbareruimte': None,
  'bagNummeraanduiding': None},
 'id': '10004',
 'idNummer': 'GL 7226',
 'serienummer': 'BAO01404',
 'clusterId': None,
 'eigenaarId': '184',
 'eigenaarNaam': 'Amsterdam Dump Locatie',
 'status': 0,
 'fractieCode': '2',
 'fractieOmschrijving': 'Glas',
 '

In [11]:
info_trash['_embedded']['container'][19]

{'_links': {'schema': 'https://schemas.data.amsterdam.nl/datasets/huishoudelijkafval/container/v2',
  'self': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/container/10254',
   'title': '10254',
   'id': '10254'},
  'cluster': None,
  'locatie': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containerlocatie/10253',
   'title': '10253',
   'id': '10253'},
  'type': {'href': 'https://api.data.amsterdam.nl/v1/huishoudelijkafval/containertype/382',
   'title': '382',
   'id': '382'},
  'bagHoofdadresVerblijfsobject': None,
  'gbdBuurt': {'href': 'https://api.data.amsterdam.nl/v1/gebieden/buurten/03630980000340',
   'title': '03630980000340',
   'identificatie': '03630980000340'},
  'bagOpenbareruimte': None,
  'bagNummeraanduiding': None},
 'id': '10254',
 'idNummer': 'GL 7144',
 'serienummer': 'BAO01529',
 'clusterId': None,
 'eigenaarId': '184',
 'eigenaarNaam': 'Amsterdam Dump Locatie',
 'status': 0,
 'fractieCode': '2',
 'fractieOmschrijving': 'Glas',
 '

In [7]:
info_trash['_embedded']['container'][0]['geometrie']['coordinates']

[125270.13808738378, 484296.416385144]

#### Iteration through pages

In [21]:
all_records_glass_trash = []
page = 1

url_trash_pagination = "https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?fractieCode=2&_pageSize=100"

while url_trash_pagination:

    print(f"Downloading page {page}")

    response_page = requests.get(url_trash_pagination)
    response_page.raise_for_status()

    info_page = response_page.json()

    embedded = info_page["_embedded"]
    collection = next(iter(embedded))

    missing_geometry = 0
    for container in embedded[collection]:
        geometrie = container.get("geometrie")
        if geometrie is None:
            missing_geometry += 1
            continue
        if geometrie and "coordinates" in geometrie:
            all_records_glass_trash.append(geometrie["coordinates"])
    
    # Get the URL for the next page (if any)
    url_trash_pagination = info_page["_links"].get("next", {}).get("href")
    print(url_trash_pagination)

    page += 1

https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=2
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=3
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=4
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=5
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=6
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=7
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=8
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=9
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=10
https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=11
https://api.data.amsterdam.n

In [17]:
url_pagination = info_page["_links"].get("next", {}).get("href")
url_pagination

'https://api.data.amsterdam.nl/v1/huishoudelijkafval/container?_pageSize=100&fractieCode=2&page=2'

#### Transform to Data Frame¶

In [22]:
len(all_records_glass_trash)

2980

In [29]:
glass_trash_data = pd.DataFrame(all_records_glass_trash)

In [24]:
glass_trash_data

,0,1
0,125270.138087,484296.416385
1,125172.998048,484171.993658
2,125258.364368,484073.697609
3,125126.999833,483976.998266
4,115581.963477,489440.459496
...,...,...
2975,123068.998130,485428.993694
2976,123258.066508,485508.365730
2977,123471.443863,485604.969668
2978,123584.995591,485665.999347


In [30]:
glass_trash_data['object'] = 'glass_trash_can'

In [34]:
glass_trash_data = glass_trash_data[['object', 0, 1]]

In [35]:
glass_trash_data

,object,0,1
0,glass_trash_can,125270.138087,484296.416385
1,glass_trash_can,125172.998048,484171.993658
2,glass_trash_can,125258.364368,484073.697609
3,glass_trash_can,125126.999833,483976.998266
4,glass_trash_can,115581.963477,489440.459496
...,...,...,...
2975,glass_trash_can,123068.998130,485428.993694
2976,glass_trash_can,123258.066508,485508.365730
2977,glass_trash_can,123471.443863,485604.969668
2978,glass_trash_can,123584.995591,485665.999347


## CSV

#### Load dataframe

In [38]:
housing_units = pd.read_csv(
    "BAG_verblijfsobject_Actueel.csv",
    sep=";",
    low_memory=False
)

In [39]:
housing_units.head()

,identificatie,aanduidingInOnderzoek,geconstateerd,heeftIn:BAG.NAG.identificatieHoofdadres,huisnummerHoofdadres,huisletterHoofdadres,huisnummertoevoegingHoofdadres,postcodeHoofdadres,ligtAan:BAG.ORE.identificatieHoofdadres,ligtAan:BAG.ORE.naamHoofdadres,...,ligtIn:GBD.GGW.code,ligtIn:GBD.GGW.naam,ligtIn:GBD.GGP.identificatie,ligtIn:GBD.GGP.code,ligtIn:GBD.GGP.naam,ligtIn:GBD.SDL.identificatie,ligtIn:GBD.SDL.code,ligtIn:GBD.SDL.naam,geometrie,bagproces
0,363010000543292,N,N,363200000006110,1,NaN,H,1015NR,363300000002529,Eerste Anjeliersdwarsstraat,...,GA01,Centrum-West,NaN,NaN,NaN,3630000000018,A,Centrum,POINT (120700 487936),NaN
1,363010000543300,N,N,363200000006119,5,NaN,1,1015NR,363300000002529,Eerste Anjeliersdwarsstraat,...,GA01,Centrum-West,NaN,NaN,NaN,3630000000018,A,Centrum,POINT (120703 487926),Muteren naar aanleiding van signalering
2,363010000543305,N,N,363200000006124,7,NaN,2,1015NR,363300000002529,Eerste Anjeliersdwarsstraat,...,GA01,Centrum-West,NaN,NaN,NaN,3630000000018,A,Centrum,POINT (120705 487922),Muteren naar aanleiding van signalering
3,363010000543308,N,N,363200000403263,17,NaN,NaN,1015NR,363300000002529,Eerste Anjeliersdwarsstraat,...,GA01,Centrum-West,NaN,NaN,NaN,3630000000018,A,Centrum,POINT (120709 487918),Muteren naar aanleiding van signalering
4,363010000543318,N,N,363200000006133,29,NaN,NaN,1015NR,363300000002529,Eerste Anjeliersdwarsstraat,...,GA01,Centrum-West,NaN,NaN,NaN,3630000000018,A,Centrum,POINT (120708 487907),Muteren naar aanleiding van signalering


In [40]:
housing_units.info()

<class 'pandas.DataFrame'>
RangeIndex: 582145 entries, 0 to 582144
Data columns (total 55 columns):
 #   Column                                   Non-Null Count   Dtype  
---  ------                                   --------------   -----  
 0   identificatie                            582145 non-null  int64  
 1   aanduidingInOnderzoek                    582145 non-null  str    
 2   geconstateerd                            582145 non-null  str    
 3   heeftIn:BAG.NAG.identificatieHoofdadres  582145 non-null  int64  
 4   huisnummerHoofdadres                     582145 non-null  int64  
 5   huisletterHoofdadres                     114673 non-null  str    
 6   huisnummertoevoegingHoofdadres           186805 non-null  str    
 7   postcodeHoofdadres                       572341 non-null  str    
 8   ligtAan:BAG.ORE.identificatieHoofdadres  582145 non-null  int64  
 9   ligtAan:BAG.ORE.naamHoofdadres           582145 non-null  str    
 10  ligtIn:BAG.WPS.identificatieHoofdadres   58

## Context

Amsterdam spans approximately 16,748 unique 6-digit postal codes (e.g., 1012 JS). These are distributed across roughly 75 unique 4-digit district areas (e.g., 1012). Each specific 6-digit postcode generally covers an average of just 15 to 20 individual addresses.

https://www.amsterdam.nl/stelselpedia/bag-index/producten-bag/prodspec-postcode/

Postal codes in Amsterdam fall within the 1000 to 1109 range. They are divided across the city's seven main districts and Weesp as follows:

- Centrum (1011 - 1019): Covers the historic inner city, canals, and the Eastern Docklands.
- Noord (1020 - 1039): Situated directly across the IJ river.
- Westpoort (1040 - 1049): Amsterdam's primary port and industrial area.
- West (1050 - 1059): Neighborhoods just west of the Singelgracht.
- Nieuw-West (1060 - 1069): Residential areas extending towards the western outskirts.
- Zuid (1070 - 1083): Includes De Pijp, Oud-Zuid, and the Zuidas business district.
- Oost (1086 - 1099): The eastern residential boroughs.
- Zuidoost (1100 - 1109): The southeastern region, including the Bijlmer and Gaasperplas.
https://nl.wikipedia.org/wiki/Lijst_van_postcodes_1000-1999_in_Nederland